source/common/network/connection_impl.cc

```diff
@@ -258,20 +277,32 @@ void ConnectionImpl::addBytesSentCallback(BytesSentCb cb) {
   bytes_sent_callbacks_.emplace_back(cb);
 }
 
-void ConnectionImpl::write(Buffer::Instance& data) {
+void ConnectionImpl::write(Buffer::Instance& data, bool end_stream) {
+  ASSERT(!end_stream || enable_half_close_);
+
+  if (write_end_stream_) {
+    // It is an API violation to write more data after writing end_stream,
+    // but a duplicate end_stream with no data is ok.
```

Comment:
When does this happen? (Duplicate end_stream with no data). More comments?

Reply:
I don't think it does happen.  But duplicate end_stream without data should be harmless, so I didn't think it was worth asserting.  But with data, we know that data will get dropped, so I thought that was worth an assert. 

Want me to expand on the comment?

Reply:
Sure I would just expand a bit and say that it shouldn't happen but is a guard just in case.